src/utils/dateUtils.ts

```diff
@@ -86,7 +86,7 @@ export function isInFuture(datestring: string): boolean {
 }
 
 export function isSameDate(first: Date, second: Date): boolean {
-  return first.toISOString().slice(0, 10) == second.toISOString().slice(0, 10);
+  return dayjs(first).isSame(dayjs(second), 'day')
```

Comment:
This change might look dangerous. The reason I did it was that doing 
`(new Date("2024-06-08T00:00:00").toISOString().slice(0, 10))` returns "2024-06-07" due to it taking time zones into account. Using `isSame` with 'day' as parameter fixed that, and to me it feels more safe to handle comparisons using a date library.